mmdet/models/plugins/msdeformattn_pixel_decoder.py

```diff
@@ -0,0 +1,303 @@
+            Defaults to dict(type='GN', num_groups=32).
+        act_cfg (:obj:`mmcv.ConfigDict` | dict): Config for activation.
+            Defaults to dict(type='ReLU').
+        encoder (:obj:`mmcv.ConfigDict` | dict): Config for transformer
+            encoder. Defaults to None.
+        positional_encoding (:obj:`mmcv.ConfigDict` | dict): Config for
+            transformer encoder position encoding. Defaults to
+            dict(type='SinePositionalEncoding', num_feats=128,
+            normalize=True).
+        init_cfg (:obj:`mmcv.ConfigDict` | dict): Initialization config dict.
+    """
+
+    def __init__(self,
+                 in_channels=[256, 512, 1024, 2048],
+                 feat_channels=256,
+                 out_channels=256,
+                 num_return_feat_levels=3,
+                 norm_cfg=dict(type='GN', num_groups=32),
+                 act_cfg=dict(type='ReLU'),
+                 encoder=dict(
+                     type='DetrTransformerEncoder',
+                     num_layers=6,
+                     transformerlayers=dict(
+                         type='BaseTransformerLayer',
+                         attn_cfgs=dict(
+                             type='MultiScaleDeformableAttention',
+                             embed_dims=256,
+                             num_heads=8,
+                             num_levels=3,
+                             num_points=4,
+                             im2col_step=64,
+                             dropout=0.0,
+                             batch_first=False,
+                             norm_cfg=None,
+                             init_cfg=None),
+                         feedforward_channels=1024,
+                         ffn_dropout=0.0,
+                         operation_order=('self_attn', 'norm', 'ffn', 'norm')),
+                     init_cfg=None),
+                 positional_encoding=dict(
+                     type='SinePositionalEncoding',
+                     num_feats=128,
+                     normalize=True),
+                 init_cfg=None):
+        super().__init__(init_cfg=init_cfg)
+        self.num_input_levels = len(in_channels)
+        self.num_encoder_feat_levels = \
+            encoder.transformerlayers.attn_cfgs.num_levels
+        assert self.num_encoder_feat_levels >= 1, \
+            'num_levels in attn_cfgs must be at least one'
+        input_proj_list = []
+        # from top to down (low to high resolution)
+        for i in range(
+                self.num_input_levels - 1,
+                self.num_input_levels - self.num_encoder_feat_levels - 1, -1):
+            input_proj = ConvModule(
+                in_channels[i],
+                feat_channels,
+                kernel_size=1,
+                norm_cfg=norm_cfg,
+                act_cfg=None,
+                bias=True)
+            input_proj_list.append(input_proj)
+        self.input_projs = ModuleList(input_proj_list)
```

Comment:
self.input_projs -> self.input_convs, input_proj -> input_conv
To be consistent with output_convs, lateral_convs